### PR TITLE
Use ? in core/std macros

### PR DESCRIPTION
**Pull request by taiki-e**

### CODE REVIEW

**Comment by camelid** [comment]
(Note that this appears to be specifically using `?` to allow optional trailing commas.)

**Comment by taiki-e** [comment]
@camelid This PR does not change any behavior, trailing commas are allowed in core/std macros since https://github.com/rust-lang/rust/pull/48056.

File: library/core/src/macros/mod.rs [hunk 1]
```diff
@@ -6,15 +6,12 @@ macro_rules! panic {
     () => (
         $crate::panic!("explicit panic")
     );
-    ($msg:literal) => (
+    ($msg:literal $(,)?) => (
```
Comment (taiki-e) [inline library/core/src/macros/mod.rs:9]: There is no `($msg:literal,) => ($panic!($msg))` in the previous code, but this seems previously covered by `($msg:expr,) => ($panic!($msg))`.

Comment (camelid) [reply library/core/src/macros/mod.rs:9]: Ah, that's what confused me then.

**Pull request merged**